Source/Peripheral.swift

```diff
@@ -471,12 +468,22 @@ public class Peripheral {
      Immediately after that `.Complete` is emitted.
      */
     public func writeValue(_ data: Data, for descriptor: Descriptor) -> Observable<Descriptor> {
-        return .create { observer in
-            let disposable = self.ensureValidPeripheralState(for: self.monitorWrite(for: descriptor).take(1))
-                .subscribe(observer)
-            self.peripheral.writeValue(data, for: descriptor.descriptor)
-            return disposable
+        let monitorWrite = self.monitorWrite(for: descriptor).take(1)
+        return ensureValidStateAndCallIfNecessary(
+            for: monitorWrite,
+            call: { self.peripheral.writeValue(data, for: descriptor.descriptor) }
+        )
+    }
+
+    func ensureValidStateAndCallIfNecessary<T>(
```

Comment:
I would like to point out that this is special case for `ensureValidPeripheralState`
and also that later call is executed only when subscription of observable is successful.

`ensureValidPeripheralState(for: callPostSubscription: )`